components/ChatApp.tsx

```diff
@@ -48,10 +48,42 @@ export class ChatApp extends React.Component<
         }
     }
 
-    public messageAdded = (message: string) => {
-        this.setState((prevState, props) => ({
+    public messageAdded = async (message: any) => {
+        this.setState((prevState) => ({
             messages: [...prevState.messages, message],
         }));
+        if (message.index > message.channel.lastConsumedMessageIndex) {
+            // % 30 is here because message index can be > 30 and when i call the method "getMessages"
+            // I only get 30 (it's configurable)
+            message.channel.updateLastConsumedMessageIndex(message.index % 30);
+        }
+    }
+
+    // factored for better lisibility for now, but I'll duplicate later, called 2 times for now
+    public updateChannels = async () => {
+        if (this.chatClient) {
+            this.setState({ channels: [] });
+            const channels = await this.chatClient.getSubscribedChannels(
+                undefined,
+            );
+            channels.items.map(async (channel) => {
+                const messages = await channel.getMessages();
+                const index = messages.items.length;
+
+                this.setState((prevState) => ({
```

Comment:
it looks like you dont even should use `for loop` suggested above but a reduce function https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/reduce